pkg/module_manager/scheduler/scheduler.go

```diff
@@ -386,13 +394,110 @@ func (s *Scheduler) getEnabledModuleNames() []string {
 	return enabledModules
 }
 
+func (s *Scheduler) getEnabledModuleNamesByOrder(weights ...node.NodeWeight) (map[node.NodeWeight][]string, error) {
+	result := make(map[node.NodeWeight][]string, 0)
+	var err error
+	switch {
+	// get modules of all weights
+	case len(weights) == 0:
+		if s.root == nil {
+			return nil, fmt.Errorf("graph is empty")
+		}
+		var (
+			dfsErr     error
+			allWeights []node.NodeWeight
+		)
+		err = graph.DFS(s.dag, s.root.GetName(), func(name string) bool {
+			vertex, props, err := s.dag.VertexWithProperties(name)
+			if err != nil {
+				dfsErr = fmt.Errorf("couldn't get %s vertex from the graph: %v", name, err)
+				return true
+			}
+
+			if props.Attributes["type"] == string(node.WeightType) {
+				allWeights = append(allWeights, vertex.GetWeight())
+			}
+
+			return false
+		})
+		if err != nil {
+			return nil, err
+		}
+		if dfsErr != nil {
+			return nil, dfsErr
+		}
+		result, err = s.getEnabledModuleNamesByOrder(allWeights...)
+
+	// get modules of specific weight
+	case len(weights) == 1:
+		// check if the weight vertex exists
+		rootVertex, err := s.dag.Vertex(weights[0].String())
+		if err != nil {
+			return nil, err
+		}
+
+		// get all modules of the weight
+		var (
+			bfsErr         error
+			enabledModules = []string{}
+		)
+		// search for all module vertices which have desired weight
+		err = graph.BFS(s.dag, rootVertex.GetName(), func(name string) bool {
+			vertex, props, err := s.dag.VertexWithProperties(name)
+			if err != nil {
+				bfsErr = fmt.Errorf("couldn't get %s vertex from the graph: %v", name, err)
+				return true
+			}
+
+			if props.Attributes["type"] == string(node.ModuleType) && vertex.GetState() {
+				if vertex.GetWeight() == rootVertex.GetWeight() {
+					enabledModules = append(enabledModules, vertex.GetName())
+					return false
+				}
+				// a vertex with different weight has been found - exit
+				return true
+			}
+
+			// a weight vertex has been found - continue
+			return false
+		})
+		if err != nil {
+			return nil, err
+		}
+		if bfsErr != nil {
+			return nil, bfsErr
+		}
+		sort.Slice(enabledModules, func(i, j int) bool {
+			return enabledModules[i] < enabledModules[j]
+		})
+		if len(enabledModules) > 0 {
+			result[rootVertex.GetWeight()] = enabledModules
+		}
+
+	// get modules of specified weights
+	default:
+		sort.Slice(weights, func(i, j int) bool { return weights[i] < weights[j] })
```

Comment:
I guess it would be nice to implement [sortable](https://pkg.go.dev/sort#Interface) for NodeWeight

Reply:
done